### PR TITLE
Angela TUG-46 Create Falcon Ansible role

### PR DESCRIPTION
**Pull request by angelafevi**
This PR is creating a new role to install and configure CrowdStrike Falcon sensor in the `Global Immutable base image` Ubuntu 18.

The sensor installed is compatible also with Ubuntu 16/20.

When deploying this new role we will need to be aligned with Security Team.

The sensor installation can be seen and tested on:
- [Falcon CrowdStrike Cloud](https://falcon.eu-1.crowdstrike.com/hosts/hosts/host/110bbc24023c40eb9577e474f9056442?filter=product_type_desc%3A%27Server%27%2Blast_seen%3A%3E%27Last%20day%27&sort=zone_group.desc) -please contact Security to gain access.
- The instance `i-0356bebaaef56d485` deployed using the image `angela-testimmutable_multienv_base`
- The jenkins job [Test - Global Immutable Base AMI](https://jenkins.stuart.com/job/Test%20-%20Global%20Immutable%20Base%20AMI/)

This is the first of two PRs needed to deploy the new role. Pending to open a PR under infra repository to define a final version of this role to be used. 
 
(All the commits will be squashed when merging)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,60 @@
-# ansible-falcon
-Ansible role to download and install CrowdStrike Falcon sensor
+Role Name
+=========
+The purpose of this role is to download and configure the Falcon to install it on our EC2 instances. The sensor must be downloaded from Crowdstrike Cloud using MFA.
```
Comment (esanchezm) [inline README.md]:
```suggestion
The purpose of this role is to download, install and configure CrowdStrike Falcon sensor it on our EC2 instances. The sensor must be downloaded from Crowdstrike Cloud using MFA.
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,60 @@
...
+* `falcon_instaler_id` - Installer ID: default version to be installed
+* `falcon_install_tmp_dir` - Where should the sensor file be downloaded to on Linux
+
+Supported secrets:
+
+This information can be found on under Falcon UI and won't be changed in the near future:
+
+* `falcon_client_id` - CrowdStrike OAUTH Client ID
+* `falcon_client_secret` - CrowdStrike OAUTH Client Secret
+* `falcon_cid` - CrowdStrike Customer ID
```
Comment (esanchezm) [inline README.md]: I think this should have a clearer name:

```suggestion
* `falcon_customer_id` - CrowdStrike Customer ID
```

File: defaults/secrets.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+          6664323335346463333433353236343134633132323739613738
+falcon_client_secret: !vault |
+          $ANSIBLE_VAULT;1.1;AES256
+          30393263343031316262616533623862656637363066663431643330346666663637643330306363
+          6565663134373832376233356563376130636634356164310a363233626432346366353738356638
+          33636165666261373965343332633337376332343663306463303330326665383365383631313130
+          6363306465396266360a313064613930333361393161383864326166626565643466616632303930
+          32303832343530323561623734386431643333333735633861663133373433636634656335616236
+          6465386538363431653731626365306564303537333330343032
+falcon_cid: !vault |
```
Comment (esanchezm) [inline defaults/secrets.yml]:
```suggestion
falcon_customer_id: !vault |
```

File: tasks/install.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+---
+## Install the package using a generic packer manager
+- name: Install Falcon Sensor Package (Linux)
+  become: true
+  package:
+    deb: "{{ falcon_sensor_pkg }}"
+    state: present
+
+- name: Configure Falcon Sensor Options
+  command:  "/opt/CrowdStrike/falconctl -s --cid={{ falcon_cid }} --provisioning-token={{ falcon_provisioning_token }}"
```
Comment (esanchezm) [inline tasks/install.yml]:
```suggestion
  command:  "/opt/CrowdStrike/falconctl -s --cid={{ falcon_customer_id }} --provisioning-token={{ falcon_provisioning_token }}"
```

**Review by esanchezm — CHANGES_REQUESTED** [review]
Small things like doc and renaming a variable to make it clearer

**Review by mjimeneznet — APPROVED** [review]
Perfect role!

File: README.md (hunk 2 above)
Comment (angelafevi) [reply README.md]: It's true, changing it since it can be misleading

**Pull request merged**